### PR TITLE
Fix Bug from Mock PE Peer Review

### LINKED ISSUE #84: [PE-D] new trip - added trip statement.
Not sure if it is intentional but the added trip statement is not very self explanatory. Like what is "...trip 0 [TRIP]...". Would be better to write in a more comprehensive way.

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: tikimonarch/ped#1

### LINKED ISSUE #85: [PE-D] cosmetics - numbering missing in list /p /d?
When listing with just list /p there is numbering.

But when listing with list /p /d there is no numbering. There should be consistency. Or maybe this is intentional then ok

Because it was written that the following is expected for both so I assume representation is same for both.

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: tikimonarch/ped#2

### LINKED ISSUE #88: [PE-D] IndexOutOfBoundsException for negative index
When `delete trip`, a negative index like `-1` and a non-number input (e.g. `help`) will give `IndexOutOfBoundsException` and `java.lang.NumberFormatException`,  and halt the program.

Same for `load trip` 

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: t170815518/ped#1

### LINKED ISSUE #89: [PE-D] Is negative currency amount reasonable?
I am not sure converting -1 foreign currency is reasonable in the app. 
```
>> convert /toSGD -1
_________________________________________________________________________
Processing... Please Wait.
That amount in your local currency is -0.20 SGD.
_____________________________________________________
```

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: t170815518/ped#2

### LINKED ISSUE #90: [PE-D] Delete place doesn't work after loaded trip
After load a trip, I am not sure why i cannot delete the place. 
```
_________________________________________________________________________
>> delete /p 1
_________________________________________________________________________
Sorry I can't find the place. Please enter a valid index.
_________________________________________________________________________
>> load trip
_________________________________________________________________________
Here are your existing trips.
1. 0 Days - NUS Lab
2. 4 Days - China
3. 3 Days - China 2

Which one do you want to load? Enter the index:delete /p 1
Exception in thread "main" java.lang.NumberFormatException: For input string: "delete /p 1"
	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.base/java.lang.Integer.parseInt(Integer.java:652)
	at java.base/java.lang.Integer.parseInt(Integer.java:770)
	at seedu.trippie.command.LoadTripCommand.execute(LoadTripCommand.java:19)
	at seedu.trippie.Trippie.executeCommand(Trippie.java:74)
	at seedu.trippie.Trippie.run(Trippie.java:49)
	at seedu.trippie.Trippie.main(Trippie.java:23)
```

-------------
Labels: `severity.High` `type.FeatureFlaw`
original: t170815518/ped#3

### LINKED ISSUE #92: [PE-D] Adding trip with empty trip name
When creating new trip, if I type "enter" without specifiying a trip name, it is accepted at the start.

It is rejected in the end, so it is likely a very low severity bug.

-------------
Labels: `severity.VeryLow` `type.FeatureFlaw`
original: TanJunHong/ped#1

### LINKED ISSUE #93: [PE-D] Adding first trip, Trip 0
When new trip is added, trip starts with number 0 instead of 1.

-------------
Labels: `severity.VeryLow` `type.FeatureFlaw`
original: TanJunHong/ped#3

### LINKED ISSUE #94: [PE-D] In loading trip, invalid index crashes the program
When using load trip, invalid index throws IndexOutOfBoundsException crashes the program. It should be handled.

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: TanJunHong/ped#4

### LINKED ISSUE #95: [PE-D] Adding a place with negative day
When adding a place to a trip, adding a negative day is allowed. 

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: TanJunHong/ped#6

### LINKED ISSUE #97: [PE-D] Listing all places with negative index
When listing a place by day, using a negative index will list all days.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: TanJunHong/ped#8

### LINKED ISSUE #98: [PE-D] Editing budget to negative budget
When editing budget, negative budget is allowed.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: TanJunHong/ped#9

### LINKED ISSUE #99: [PE-D] Adding an expense of negative day and value
When adding an expense, negative day and value is allowed.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: TanJunHong/ped#10

### LINKED ISSUE #100: [PE-D] Typo in "exchange" in quick start
Exchange rate is spelt as exhange rate.

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: TanJunHong/ped#11

### LINKED ISSUE #104: [PE-D] The cost of expense does not mention currency type.
When you add the cost of the item you bought, no where is it mention if the currency you want the user to add is in sgd or the foreign currency. It might confuse the users. Even though you mention it in the listing all expenses it might be better to specify it in the buy command section of the user guide and maybe when you specifiy the format of the buy command in the app (help section). 

-------------
Labels: `severity.Medium` `type.DocumentationBug`
original: sugandha929/ped#4

### LINKED ISSUE #106: [PE-D] Currency type says $xxx eur
The currency in the list says $2.00 eur. This is an incorrect format for currency. This might also confuse users as to which currency type you are reffering to.

-------------
Labels: `severity.Medium` `type.DocumentationBug`
original: sugandha929/ped#6

### LINKED ISSUE #110: [PE-D] Viewing Budget
Would be great if i could my budget in a trip even if i have an empty expenses list. As for now if my expenses list is empty there is no way i can view my budget. 

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: shreytheshreyas/ped#4

### PR DESCRIPTION
**Pull request by ShawnTanzc**
Fix #84 Fix #85 Fix #88 Fix #89 Fix #90 Fix #92 Fix #93 Fix #94
Fix #95 Fix #97 Fix #98 Fix #99 Fix #100 Fix #104 Fix #106 Fix #110

### CODE REVIEW

File: src/main/java/seedu/trippie/command/LoadTripCommand.java [hunk 1]
```diff
@@ -20,6 +20,9 @@ public void execute(Ui ui, TrippieData trippieData) {
             try {
                 System.out.print("Which one do you want to load? Enter the index:");
                 index = Integer.parseInt(ui.getLine());
+                if (index < 0) {
+                    throw new IndexOutOfBoundsException("Please enter a valid trip number");
+                }
```
Comment (weishuangtan) [inline src/main/java/seedu/trippie/command/LoadTripCommand.java:25]: Already resolved in Issue #82, don't need to add as IndexOutOfBoundsException will already be thrown when loadCurrentTripFromFile() cannot access the index.

**Review by weishuangtan — APPROVED** [review]
Thanks Shawn for fixing most of the issues! Other than the minor changes, LGTM! 👍

File: src/main/java/seedu/trippie/command/AddPlaceCommand.java [hunk 2]
```diff
@@ -43,13 +44,16 @@ public AddPlaceCommand(String userInput) throws TrippieInvalidArgumentException
...
+        return inputWithoutDay.split(" /n ")[1];
     }
 
-    public int extractDay(String userInput) {
-        String withoutTime = userInput.split(" /t ")[0];
-        String day = withoutTime.split(" /d ")[1];
+    public int extractDay(String userInput) throws TrippieInvalidArgumentException {
+        String inputWithoutTime = userInput.split(" /t ")[0];
+        String day = inputWithoutTime.split(" /d ")[1];
+        if (Integer.parseInt(day.trim()) < 0) {
```
[the remark this answers is not in the document]
Comment (ShawnTanzc) [reply src/main/java/seedu/trippie/command/AddPlaceCommand.java]: Roger

File: src/main/java/seedu/trippie/command/LoadTripCommand.java (hunk 1 above)
Comment (ShawnTanzc) [reply src/main/java/seedu/trippie/command/LoadTripCommand.java]: Roger

File: src/main/java/seedu/trippie/command/AddExpenseCommand.java [hunk 3]
```diff
@@ -37,22 +39,28 @@ public AddExpenseCommand(String userInput) throws TrippieInvalidArgumentExceptio
...
+            throw new TrippieInvalidArgumentException(NEGATIVE_COST_MESSAGE);
+        }
         return Float.parseFloat(expenseCost);
 
     }
 
-    public int extractDayBought(String userInput) {
+    public int extractDayBought(String userInput) throws TrippieInvalidArgumentException {
         String onlyDay = userInput.split(" /d ")[1];
+        if (Integer.parseInt(onlyDay) < 0) {
```
[the remark this answers is not in the document]
Comment (ShawnTanzc) [reply src/main/java/seedu/trippie/command/AddExpenseCommand.java]: Roger

File: docs/UserGuide.md [hunk 4]
```diff
@@ -75,7 +75,7 @@ Written below are the steps to be taken to set up the application:
     * If your operating system is MacOS, use Terminal.
 6. Enter `java -jar trippie.jar` in your command line tool and press enter.
 7. If the setup is correct, you should see a welcome message. Then, it is now ready to go.
-8. Create your first trip by entering [`new trip`](#51-creating-a-new-trip-new-trip-ivander) to the command line. Then, enter your trip name, start date, exhange rate, currency, and budget.
+8. Create your first trip by entering [`new trip`](#51-creating-a-new-trip-new-trip-ivander) to the command line. Then, enter your trip name, start date, exchange rate, currency, and budget.
```
Comment (felixhalim) [inline docs/UserGuide.md:78]: Good catch!

File: src/main/java/seedu/trippie/data/Expense.java [hunk 5]
```diff
@@ -25,7 +25,7 @@ public int getExpenseDayBought() {
 
     @Override
     public String toString() {
-        return "Day " + getExpenseDayBought() + ": " + getExpenseName() + " - $"
+        return "Day " + getExpenseDayBought() + ": " + getExpenseName() + " - "
                 + String.format("%.2f",getExpenseCost());
```
Comment (ivanderjmw) [inline src/main/java/seedu/trippie/data/Expense.java:29]: Fixes #106

**Review by ivanderjmw — APPROVED** [review]
Thank you Shawn! Great bug fixes. LGTM.

File: src/main/java/seedu/trippie/command/AddPlaceCommand.java [hunk 6]
```diff
@@ -105,7 +109,7 @@ public void execute(Ui ui, TrippieData trippieData) {
     }
 
     public void sortPlaceList(List<Place> sortedPlaces) {
-        Boolean swapped = false;
+        boolean swapped = false;
```
Comment (felixhalim) [inline src/main/java/seedu/trippie/command/AddPlaceCommand.java:112]: Perhaps name the variable to `isSwapped` ?

**Review by felixhalim — APPROVED** [review]
LGTM

**Pull request merged**